pyup/requirements.py

```diff
@@ -229,6 +230,34 @@ def get_requirement_class(self):   # pragma: no cover
         return Requirement
 
 
+class SetupPyFile(RequirementFile):
+    def _parse(self):
+        self._requirements, self._other_files = [], []
+
+        Klass = self.get_requirement_class()
+
+        def setup(*args, **kwargs):
+            reqs = kwargs.get('install_requires', []) + kwargs.get('tests_require', [])
+
+            for num, line in enumerate(reqs):
+                req = Klass.parse(line, num)
+
+                if req.package:
+                    self._requirements.append(req)
+
+        try:
+            module = __import__('setuptools')
+        except ImportError:
+            module = __import__('distutils.core')
+
+        old_setup = module.setup
+        module.setup = setup
+
+        exec(self.content, globals())
```

Comment:
I'm afraid we can't to that as it allows to execute arbitrary python :(

The code runs on [pyup.io](https://pyup.io) and if someone crafts a `setup.py` that runs some malicious python code, he/she can basically take over the service.